var/spack/repos/builtin/packages/eccodes/package.py

```diff
@@ -317,7 +320,7 @@ def cmake_args(self):
             self.define("ENABLE_JPG_LIBJASPER", jp2k == "jasper"),
             self.define("ENABLE_JPG_LIBOPENJPEG", jp2k == "openjpeg"),
             self.define_from_variant("ENABLE_PNG", "png"),
-            self.define_from_variant("ENABLE_AEC", "aec"),
+            self.define("ENABLE_AEC", "aec" if "+aec" in self.spec else "OFF"),
```

Comment:
Why did this line change (`aec` is a variant)?

Reply:
Followed the pattern of BUILD_SHARED_LIBS:
`self.define("BUILD_SHARED_LIBS", "BOTH" if "+shared" in self.spec else "OFF"),`
because to force the build without aec we now need to define -DENABLE_AEC=OFF

I wasn't sure whether `define_from_variant` where variant is boolean, would result in an 'ON'/'OFF' value for `ENABLE_AEC`?

Reply:
Otherwise thanks for the review!

Reply:
It does, in fact "better", since it adds the type: `-DENABLE_AEC:BOOL=ON` automatically.

Reply:
Ok, thanks a lot for the explanation. Returned to `define_from_variant` in https://github.com/spack/spack/pull/40564/commits/c41159b46734e512267c7c0b5aa90b7edba193aa